OR || operation not returning first non-zero operation with integers

**V version:** V 0.1.29 0f1c484

**OS:** Linux

**What did you do?**
```
println(2 || 1)
```

**What did you expect to see?**
```
2
```
 
**What did you see instead?**
```
1
```

**Note**
The output is 1 due to how C works, it returns a boolean on these operations, (`0` == `false`, `1` == `true`):
`0 || 0` => `0`
`2 || 0` => `1`
`0 || 3` => `1`
`2 || 3` => `1`

`0 && 0` => `0`
`2 && 0` => `0`
`0 && 3` => `0`
`2 && 3` => `1`

Go and Rust disallow this. The question should if V should allow this or not.

- Go
`./prog.go:10:16: invalid operation: a || b (operator || not defined on int)`

- Rust

```
error[E0308]: mismatched types
 --> src/main.rs:2:20
  |
2 |     println!("{}", 3 || 4)
  |                    ^ expected `bool`, found integer

error[E0308]: mismatched types
 --> src/main.rs:2:25
  |
2 |     println!("{}", 3 || 4)
  |                         ^ expected `bool`, found integer
```

Even `3 && 4` give `1` but ideally it should give `4` or an error!?

Even in this case, Go and Rust throws an error.

- Go
`./prog.go:10:16: invalid operation: a && b (operator && not defined on int)`

- Rust

```
error[E0308]: mismatched types
 --> src/main.rs:2:20
  |
2 |     println!("{}", 3 && 4)
  |                    ^ expected `bool`, found integer

error[E0308]: mismatched types
 --> src/main.rs:2:25
  |
2 |     println!("{}", 3 && 4)
  |                         ^ expected `bool`, found integer
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -771,6 +771,14 @@ pub fn (mut c Checker) infix_expr(mut infix_expr ast.InfixExpr) table.Type {
 			return table.void_type
 		}
 		else {
+			if infix_expr.op in [.and, .logical_or] {
+				if infix_expr.left_type != table.bool_type_idx {
+					c.error('left operand for `$infix_expr.op` is not a boolean', infix_expr.left.position())
+				}
+				if infix_expr.right_type != table.bool_type_idx {
+					c.error('right operand for `$infix_expr.op` is not a boolean', infix_expr.right.position())
+				}
+			}
 			// use `()` to make the boolean expression clear error
 			// for example: `(a && b) || c` instead of `a && b || c`
 			if infix_expr.op in [.logical_or, .and] {

--- a/vlib/v/checker/tests/infix_err.out
+++ b/vlib/v/checker/tests/infix_err.out
@@ -32,8 +32,22 @@ vlib/v/checker/tests/infix_err.vv:13:9: error: `+` cannot be used with `?int`
    13 | _ = g() + int(3)
       |         ^
    14 | _ = g() + 3
+   15 |
 vlib/v/checker/tests/infix_err.vv:14:9: error: `+` cannot be used with `?int`
    12 | _ = int(0) + g() // FIXME not detected
    13 | _ = g() + int(3)
    14 | _ = g() + 3
       |         ^
+   15 |
+   16 | // binary operands
+vlib/v/checker/tests/infix_err.vv:17:5: error: left operand for `&&` is not a boolean
+   15 |
+   16 | // binary operands
+   17 | _ = 1 && 2
+      |     ^
+   18 | _ = true || 2
+vlib/v/checker/tests/infix_err.vv:18:13: error: right operand for `||` is not a boolean
+   16 | // binary operands
+   17 | _ = 1 && 2
+   18 | _ = true || 2
+      |             ^

--- a/vlib/v/checker/tests/infix_err.vv
+++ b/vlib/v/checker/tests/infix_err.vv
@@ -12,3 +12,7 @@ _ = 4 + g()
 _ = int(0) + g() // FIXME not detected
 _ = g() + int(3)
 _ = g() + 3
+
+// binary operands
+_ = 1 && 2
+_ = true || 2

